modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -344,7 +347,18 @@ public void addTempAttachment(
 
 	@Override
 	public void checkKBArticles() throws PortalException {
-		_checkKBArticles(new Date());
+		Date date = new Date();
+
+		_checkKBArticlesByExpirationDate(date);
+
+		if (_previousCheckDate == null) {
+			_previousCheckDate = new Date(
+				date.getTime() - _getKBArticleCheckInterval());
+		}
```

Comment:
Can we remove `previousCheckDate`? For what I've seen, it is only used in the query, but I don't think it is strictly necessary. It seems wrong to keep mutable state in a singleton service.

Reply:
After internal discussion, it seems not an easy thing to solve. We can think of a better solution later.